### ISSUE TITLE
Consider migrating suit logic to binary flags

### ISSUE THREAD

**Issue by jparkhouse** [jparkhouse/hanabi-tracker#13]
This might look something like 1111100 for no variant and 1111101 for blacks. Then the colour information could be constructed from which suits are turned on?

This would unlock varying the number of suits, and potentially make custom games or adding additional suits easier to manage

**Comment by Alex-Stagg** [comment]
For reference, here's how you do bitflags in TS

**Comment by YuiYukihira** [comment]
Bitflags implementation as discussed:
```
Variant Flags
--------------
R G B W Y M Bl
1 1 1 1 1 1 0 - Normal suits + rainbows
1 1 1 1 1 0 1 - Normal suits + blacks
1 1 1 1 1 1 1 - Normal suits + rainbows + blacks
0 1 1 1 1 0 0 - Normal suits - reds


Hint
----
R G B W Y M BL 1 - Positive hint
1 0 0 0 0 1 0  - Red colour hint

To include rainbows when enabled:
  R G B W Y M Bl
  1 0 0 0 0 0 0  - Start with solid colour hint
| 0 0 0 0 0 1 0  = 1 1 1 1 1 1 0 & 0 0 0 0 0 1 0 - Bitwise and of variant flags & rainbow flag
= 1 0 0 0 0 1 0  - Bitwise or of hint and result, rainbow only included if enabled


Card Knowledge 1 - Known info
--------------
R G B W Y M Bl
0 0 0 0 0 0 0  - Nothing known, all possibilities
1 0 0 0 0 1 0  - Card is either red or rainbow

To apply hint:
  R G B W Y M Bl
  0 0 0 0 0 0 0 - Nothing known
| 1 0 0 0 0 1 0 - Red or rainbow hint given
= 1 0 0 0 0 1 0 - Bitwise or, Card is either red rainbow

To determine negative info:
  R G B W Y M Bl
  1 0 0 0 0 1 0 - Card is either red or rainbow
! 0 1 1 1 1 0 1 - Bitwise not, 1's are colours we know the card is not

If the card was not selected when hint given:
  R G B W Y M Bl
  0 0 0 0 0 0 0 - We know nothing about the card
| 0 1 1 1 1 0 1 = ! 1 0 0 0 0 1 0 - Bitwise not of the clue information
= 0 1 1 1 1 0 1 - Card is not red or rainbow, so must be green, blue, yellow or black.

^ Let's say black's aren't enabled though
  R G B W Y M Bl
  0 1 1 1 1 0 1
& 1 1 1 1 1 1 0 - Do a bitwise and of the enabled suits
= 0 1 1 1 1 0 0 - Card is either green, blue, white or yellow




Hinting
----

hintWithRainbow = hint | (enabledVariants & rainbow)

for card in hand:
  if card.selected:
    card.colours = (card.colours | hintWithRainbow) & enabledVariants
  else:
    card.colours = (card.colours | !hintWithRainbow) & enabledVariants
```

**Comment by jparkhouse** [comment]
closed with release 1.0.0
